internal/expr/prepare.go

```diff
@@ -256,24 +238,36 @@ func prepareOutput(ti typeNameToInfo, p *outputPart) ([]fullName, []outputDest,
 				outCols = append(outCols, c)
 				outDests = append(outDests, outputDest{info.structType, f})
 			}
-			return outCols, outDests, nil
 		}
 	}
 
-	// Case 2: None star target cases e.g. "...(&P.name, &P.id)".
+	// Case 2: Non-star target cases e.g. "(&P.name, &P.id)" or "name_1 AS &P.name".
+	if p.target[0].name != "*" {
+		for _, t := range p.target {
+			info, ok = ti[t.prefix]
+			if !ok {
+				return nil, nil, fmt.Errorf(`type %s unknown, have: %s`, t.prefix, strings.Join(getKeys(ti), ", "))
+			}
+
+			f, ok := info.tagToField[t.name]
+			if !ok {
+				return nil, nil, fmt.Errorf(`type %s has no %q db tag`, info.structType.Name(), t.name)
+			}
 
-	// Case 2.1: Explicit columns e.g. "name_1 AS P.name".
-	if len(p.source) > 0 {
-		for _, c := range p.source {
-			outCols = append(outCols, c)
+			outDests = append(outDests, outputDest{info.structType, f})
+
+			// Case 2.2: No columns e.g. "(&P.name, &P.id)".
+			if len(p.source) == 0 {
+				outCols = append(outCols, fullName{name: t.name})
+			}
 		}
-		return outCols, outDests, nil
```

Comment:
We don't want to remove this return, otherwise we will do both case 2.2 and 2.1, they wont be mutually exclusive

Reply:
Modified the format of `prepareOutput()` to have explicit returns for cases 1 and 2. Case 1 now returns at the end of its if statement. Case 2 returns at the end of the method. Cases 2.1 and 2.2 independently populate `outCols`, and the same return statement applies to them which is at the end of the method.